src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -107,10 +112,19 @@ export class ItemChildrenEditComponent implements OnChanges {
 
   resetScoreWeight(): void {
     this.data = this.data.map(c => ({ ...c, scoreWeight: DEFAULT_SCORE_WEIGHT }));
+    this.onChildAdded();
     this.onScoreWeightChange();
   }
 
   onScoreWeightChange(): void {
     this.childrenChanges.emit(this.data);
   }
+
+  onChildAdded(): void {
```

Comment:
You call in `onRemove` so it is not really `onChildAdded`.
`onChildrenListUpdate"? Actually as it is a function and not really a button callback I would call it by what it does, so "recomputeAddedIds". 
... but it must be called together with the data update, not forgetting any case, otherwise you would have an inconsistent state... I think that would be cleaner to just use the map in the template.

Reply:
I prefer onChildrenListUpdate as it common pattern, also it could be extended in future for another on updated operations.